addon/-private/resources/errors.ts

```diff
@@ -5,3 +5,11 @@ export class IdRequiredError extends Error {
     );
   }
 }
+
+export class IdTypeError extends TypeError {
+  constructor(modelName: string) {
+    super(
+      `While trying to request a resource from ${modelName}, the ID was of invalid type: only string and number are supported`
```

Comment:
I think it'd be helpful to also print the ID and `typeof` id here as well (just in case)

Reply:
I've added `${typeof id}`. Adding `${id}` seems to be unhelpful when `id` is neither string nor number.